integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala

```diff
@@ -493,4 +493,20 @@ object PreAggregateUtil {
     updatedPlan
   }
 
+  def createChildSelectQuery(tableSchema: TableSchema): String = {
```

Comment:
Instead of creating a SQL string, you can construct a dataframe directly and return to startCompactionForDataMap

Reply:
This sql is created to select data from the child table that will be inserted in that table itself. Load UDF would then be added to it and then a DataFrame would be created.

Reply:
I understand what you are doing, but I think there is better way to create the dataframe directly instead of going through parser again.